sw/device/tests/sim_dv/keymgr_key_derivation.c

```diff
@@ -0,0 +1,275 @@
+                        KMAC_PREFIX_0_REG_OFFSET + i * 4, kKmacPrefix[i]);
+  }
+}
+
+/** Key manager configuration steps performed in mask ROM. */
+rom_error_t keymgr_rom_test(void) {
+  ASSERT_OK(keymgr_state_check(kKeymgrStateReset));
+  keymgr_sw_binding_set(&kBindingValueRomExt, &kBindingValueRomExt);
+  keymgr_creator_max_ver_set(kMaxVerRomExt);
+  SEC_MMIO_WRITE_INCREMENT(kKeymgrSecMmioSwBindingSet +
+                           kKeymgrSecMmioCreatorMaxVerSet);
+  sec_mmio_check_values(/*rnd_offset=*/0);
+  sec_mmio_check_counters(/*expected_check_count=*/1);
+  return kErrorOk;
+}
+
+/** Key manager configuration steps performed in ROM_EXT. */
+rom_error_t keymgr_rom_ext_test(void) {
+  ASSERT_OK(keymgr_state_check(kKeymgrStateReset));
+
+  const uint16_t kEntropyReseedInterval = 0x1234;
+  ASSERT_OK(keymgr_init(kEntropyReseedInterval));
+  SEC_MMIO_WRITE_INCREMENT(kKeymgrSecMmioInit);
+  sec_mmio_check_values(/*rnd_offset=*/0);
+
+  keymgr_advance_state();
+  ASSERT_OK(keymgr_state_check(kKeymgrStateInit));
+  LOG_INFO("Keymgr entered Init State");
+
+  keymgr_advance_state();
+  ASSERT_OK(keymgr_state_check(kKeymgrStateCreatorRootKey));
+  LOG_INFO("Keymgr entered CreatorRootKey State");
+
+  // The software binding register lock is reset after advancing the key
+  // manager, so we need to call this function to update sec_mmio expectation
+  // table.
+  keymgr_sw_binding_unlock_wait();
+  sec_mmio_check_values(/*rnd_offset=*/0);
+
+  keymgr_sw_binding_set(&kBindingValueBl0, &kBindingValueBl0);
+  keymgr_owner_int_max_ver_set(kMaxVerBl0);
+  SEC_MMIO_WRITE_INCREMENT(kKeymgrSecMmioSwBindingSet +
+                           kKeymgrSecMmioOwnerIntMaxVerSet);
+  sec_mmio_check_values(/*rnd_offset=*/0);
+
+  keymgr_advance_state();
+  ASSERT_OK(keymgr_state_check(kKeymgrStateOwnerIntermediateKey));
+  LOG_INFO("Keymgr entered OwnerIntKey State");
+
+  sec_mmio_check_counters(/*expected_check_count=*/5);
+  return kErrorOk;
+}
+
+bool test_main(void) {
+  dif_rstmgr_t rstmgr;
+  dif_rstmgr_reset_info_bitfield_t info;
+  rom_error_t result = kErrorOk;
+
+  // This test is expected to run in DEV, PROD or PROD_END states.
+  LOG_INFO("lifecycle state: 0x%x", lifecycle_raw_state_get());
+
+  // Initialize pwrmgr
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &info));
+
+  // POR reset
+  if (info == kDifRstmgrResetInfoPor) {
+    LOG_INFO("Powered up for the first time, program flash");
+
+    // Initialize flash
+    init_flash();
+
+    // Lock otp secret partition
+    lock_otp_secret_partition2();
+
+    // reboot device
+    CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
+    CHECK_DIF_OK(dif_rstmgr_software_device_reset(&rstmgr));
+
+    // wait here until device reset
+    wait_for_interrupt();
+
+  } else {
+    CHECK(info == kDifRstmgrResetInfoSw);
```

Comment:
nit: might be worth printing the actual `info` value for ease of debug.

Reply:
Done

Reply:
This check is now redundant, since the else clause already qualifies the value.

Reply:
oops, I thought I removed it. Thanks